crates/telio-proxy/src/proxy.rs

```diff
@@ -523,6 +531,9 @@ impl Runtime for StateEgress {
             (pk, _, _)= update => {
                 self.replace_socket(pk).await;
             },
+            else => {
+                telio_log_warn!("StateEgress: no events to wait on ...");
+            },
```

Comment:
I think this is not tested:

```
$ git diff
diff --git a/crates/telio-proxy/src/proxy.rs b/crates/telio-proxy/src/proxy.rs
index 83b84844..7b1e1693 100644
--- a/crates/telio-proxy/src/proxy.rs
+++ b/crates/telio-proxy/src/proxy.rs
@@ -531,9 +531,9 @@ impl Runtime for StateEgress {
             (pk, _, _)= update => {
                 self.replace_socket(pk).await;
             },
-            else => {
-                telio_log_warn!("StateEgress: no events to wait on ...");
-            },
+            // else => {
+            //     telio_log_warn!("StateEgress: no events to wait on ...");
+            // },
         }
         Self::next()
     }
$ cargo test
    Finished test [unoptimized + debuginfo] target(s) in 0.25s
     Running unittests src/lib.rs (/Users/tomaszklak/Development/libtelio/target/debug/deps/telio_proxy-7ef99710b91e3a47)

running 5 tests
test proxy::tests::test_single_endpoint ... ok
test proxy::tests::test_multiple_endpoints ... ok
test proxy::tests::test_graceful_shutdown ... ok
test proxy::tests::test_muting_unmuting ... ok
test proxy::tests::test_error_handling ... ok

test result: ok. 5 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 4.02s

   Doc-tests telio-proxy

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
```

This is a pretty safe change, so if it's not possible to test this or it's very hard feel free to skip it, but if possible try to add a test that also verifies the egress case.

Reply:
I cannot think of any easy modifications on this, because of `Either<T>` future. It will always have sth to wait on :smile: . So maybe skipping this one ...